src/service/gtfs-flex-service.ts

```diff
@@ -13,35 +16,18 @@ class GtfsFlexService implements IGtfsFlexService {
     }
 
     async getAllGtfsFlex(params: FlexQueryParams): Promise<GtfsFlexDTO[]> {
-        // get a gtfsFlex repository to perform operations with gtfsFlex
-        const gtfsFlexRepository = AppDataSource.getRepository(FlexVersions);
-
-        //Set defaults if not provided
-        if (params.page_no == undefined) params.page_no = 1;
-        if (params.page_size == undefined) params.page_size = 10;
-        let skip = params.page_no == 1 ? 0 : (params.page_no - 1) * params.page_size;
-        let take = params.page_size > 50 ? 50 : params.page_size;
-
-        let where: FindOptionsWhere<FlexVersions> = {};
-
-        if (params.flex_schema_version) where.flex_schema_version = Equal(params.flex_schema_version);
-        if (params.tdei_org_id) where.tdei_org_id = Equal(params.tdei_org_id);
-        if (params.tdei_record_id) where.tdei_record_id = Equal(params.tdei_record_id);
-        if (params.tdei_service_id) where.tdei_service_id = Equal(params.tdei_service_id);
-        if (params.date_time && Utility.dateIsValid(params.date_time)) where.valid_to = Raw((alias) => `${alias} > :date`, { date: params.date_time });
-
-        // load a gtfsFlex by a given gtfsFlex id.
-        const gtfsFlex = await gtfsFlexRepository.find({
-            where: where,
-            order: {
-                tdei_record_id: "DESC",
-            },
-            skip: skip,
-            take: take,
-        });
+        //Builds the query object. All the query consitions can be build in getQueryObject()
+        let queryObject = params.getQueryObject();
+
+        let queryConfig = <QueryConfig>{
+            text: queryObject.getQuery(),
+            values: queryObject.getValues()
+        }
+
+        let result = await dbClient.query(queryConfig);
 
         let list: GtfsFlexDTO[] = [];
-        gtfsFlex.forEach(x => {
+        result.rows.forEach(x => {
 
             let flex: GtfsFlexDTO = Utility.copy<GtfsFlexDTO>(new GtfsFlexDTO(), x);;
```

Comment:
cant we use @Prop nomenclature here for conversion?

Reply:
Taken care